dist-persist/wbstack/src/Internal/PreApiWbStackUpdate.php

```diff
@@ -16,9 +16,45 @@ public function execute() {
 		@ini_set( 'memory_limit', '-1' ); // also try to disable the memory limit? Is this even a good idea?
 
 		// Run update.php
+		$domain = $GLOBALS[WBSTACK_INFO_GLOBAL]->requestDomain;
 		$mwPath = realpath( __DIR__ . '/../../../' );
-		$cmd = 'WBS_DOMAIN=' . $GLOBALS[WBSTACK_INFO_GLOBAL]->requestDomain . ' php ' . $mwPath . '/maintenance/update.php --quick';
-		exec($cmd, $out, $return);
+		$cmd = 'WBS_DOMAIN=' . $domain . ' php ' . $mwPath . '/maintenance/update.php --quick';
+
+		$stdout = fopen( 'php://stdout', 'w' );
+		$stderr = fopen( 'php://stderr', 'w' );
+
+		fwrite( $stdout, "DOMAIN: " . $domain . "\n" );
+
+		$spec = [ [ 'pipe', 'r' ], [ 'pipe', 'w' ],[ 'pipe', 'w' ] ];
+		$proc = proc_open( $cmd, $spec, $pipes );
+
+		$stdinProc = $pipes[0];
+		$stdoutProc = $pipes[1];
+		$stderrProc = $pipes[2];
+
+		$out = [];
+
+		$pid = ( proc_get_status( $proc ) )[ 'pid' ];
+		fwrite( $stdout, "PID: " . $pid . "\n" );
+
+		while( $line = fgets( $stdoutProc ) ) {
+			$line = rtrim( $line );
+			fwrite( $stdout, $line );
+			array_push( $out, $line );
+		}
+
+		while( $line = fgets( $stderrProc ) ) {
```

Comment:
It's a little unclear to me if this means that we end up batching up all the internal stdout lines and output each of those to the main stdout (and also store in `$out`) before only then addressing the stderr batch.

e.g. if we have a command that runs for an hour and the first thing to happen is something on STDERR would we see nothing at all until STDOUT has got to the end of the stream. I suspect we might find this a little misleading if it's the case.

My attempts to test this but putting something like:
```
                $stdout = fopen( 'php://stderr', 'w');
		fwrite( $stdout, “hello, world\n”);
		fclose($stdout);
```
in `update.php` simply resulted in having no output of at all from the inner script

Reply:
I sadly only can confirm your assumption. I tried to find a way around it but not successfully. For example, I wrapped the two while() loops in a `while(proc_get_status( $proc ) )[ 'running' ])` and turned the `while($line = fgets( ...` into `if` statements. And while that alternates between writing stdout and stderr output, it doesn't make sure that you get everything in the right order and (even worse) that everything at all.